docs/operator-bundle/testing-and-releasing.md

```diff
@@ -0,0 +1,181 @@
+    ```
+8. Push the index to the registry:
+    ```sh
+    docker push xtianregistry.azurecr.io/operatorhubio/azure-service-operator-index:test1
+    ```
+
+### Create your test [CatalogSource](https://docs.openshift.com/container-platform/4.7/rest_api/operatorhub_apis/catalogsource-operators-coreos-com-v1alpha1.html)
+
+The catalog source will point to your new index image:
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: xtian-test
+  namespace: olm
+spec:
+  displayName: Source for testing ASO
+  image: xtianregistry.azurecr.io/operatorhubio/azure-service-operator-index:test1
+  publisher: xtian
+  sourceType: grpc
+```
+
+(Replace the registry fqdn and names with your ones.)
+
+You should see that the catalog operator has started a pod from the index image. You can watch it interacting with the index pod using:
+```sh
+k logs -n olm deploy/catalog-operator -f
+```
+
+### Create your [Subscription](https://docs.openshift.com/container-platform/4.7/rest_api/operatorhub_apis/subscription-operators-coreos-com-v1alpha1.html)
+
+This will trigger the OLM to install the operator from the test catalog.
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  labels:
+    operators.coreos.com/azure-service-operator.operators: ""
+  name: my-azure-service-operator
+  namespace: operators
+spec:
+  channel: stable
+  name: azure-service-operator
+  source: xtian-test
+  sourceNamespace: olm
+  installPlanApproval: Manual
+  startingCSV: azure-service-operator.v0.37.0
+```
+
+(Adjust for your catalog source and starting version.)
+
+The `installPlanApproval: Manual` setting tells OLM to create an unapproved [InstallPlan](https://docs.openshift.com/container-platform/4.7/rest_api/operatorhub_apis/installplan-operators-coreos-com-v1alpha1.html) that needs to be approved manually before that version of the operator will be installed.
+This plus the `startingCSV` field pointing at the previous version of the operator will let you see the upgrade process at each step.
+
+### Approve installation of the old version
+
+You should see an unapproved InstallPlan created in the `operators` namespace for the old version of the operator.
+```sh
+k get -n operators installplans
+```
+
+Edit or patch it to set `spec.approved` to `true`.
+```sh
+k edit -n operators installplans install-abcde
+```
+
+The previous version of the operator will install and start up. You can check progress by looking at the yaml for the install plan and [ClusterServiceVersion](https://docs.openshift.com/container-platform/4.7/rest_api/operatorhub_apis/installplan-operators-coreos-com-v1alpha1.html), the logs from the catalog operator, and eventually the state of the pods in the `operators` namespace.
+
+### Approve installation of the new version
+
+Once the old version of the operator is installed, the OLM will create another install plan to upgrade it to the new version.
```

Comment:
Automatically? No need to do anything? neat

Reply:
Yeah, it's pretty cool when it works!